freqtrade/strategy/interface.py

```diff
@@ -395,7 +395,8 @@ def custom_stoploss(self, pair: str, trade: Trade, current_time: datetime, curre
         """
         return self.stoploss
 
-    def custom_entry_price(self, pair: str, current_time: datetime, proposed_rate: float,
+    def custom_entry_price(self, pair: str, trade: Trade,
```

Comment:
```suggestion
    def custom_entry_price(self, pair: str, trade: Optional[Trade],
```

Not doing this will make people believe it's always given (also the editor will believe this, leading to mistakes in strategy development).